### ISSUE TITLE
Implement SocketPlugin

### ISSUE THREAD

**Issue by bertfreudenberg** [bertfreudenberg/SqueakJS#40]
So far, there is no networking support (except via the JSBridge). It would be awesome if socket connections could be opened, but it's not quite clear how this could work. WebSockets are a possibility but are hampered by the browser's security restriction. Possibly a proxy server is needed.

**Comment by ccrraaiigg** [comment]
Yes, a proxy server is needed. I'm writing WebSockets support for Squeak on all host platforms, including SqueakJS.

**Comment by bertfreudenberg** [comment]
As of 94c003a3fbf28aba91f20631fe75d843ee169960 we have a client-side only SocketPlugin using a public CORS proxy. It only supports http/https requests. So I think a WebSocket-based client+server solution is still needed, to enable non-http connections.
Do we want to ship two different SocketPlugins? Or merge them into one?

**Comment by ccrraaiigg** [comment]
I think we should merge them into one. The networking support I mentioned is now part of another project that includes remote messaging, screen-sharing, and a new way of installing native Squeak. While those special cases have priority over transparent generalized socket use (similar to the image-updating case), I estimate having something separable in August.

**Comment by bertfreudenberg** [comment]
Awesome! I'm really busy atm, so can't review in detail, sorry. But as to your questions:

* Yes, we should use `argCount` consistently. If the primitive checks that `argCount` is the expected value then that is unnecessary, but it also doesn't cost much.
*  `fetch` vs `XMLHttpRequest` is just historical I think. We started out `XMLHttpRequest` but then `fetch` had some advantages, I think regarding CORS. It would be fine to get rid of `XMLHttpRequest`
* single PR would be fine

**Comment by ErikOnBike** [comment]
I think this issue can be closed. If getting rid of XMLHttpRequest is relevant, maybe we should create separate issue for it. (Would clean things up in SocketPlugin ;-).
